order/manager.go

```diff
@@ -142,30 +152,75 @@ func (m *Manager) PrepareOrder(ctx context.Context, order Order,
 		return nil, err
 	}
 
-	// Grab the additional information needed from our local node.
-	nextMultiSigKey, err := m.cfg.Wallet.DeriveNextKey(
-		ctx, int32(keychain.KeyFamilyMultiSig),
-	)
-	if err != nil {
-		return nil, fmt.Errorf("unable to derive "+
-			"multi sig key: %v", err)
-	}
-	order.Details().MultiSigKeyLocator = nextMultiSigKey.KeyLocator
-	var multiSigKey [33]byte
-	copy(multiSigKey[:], nextMultiSigKey.PubKey.SerializeCompressed())
-	info, err := m.cfg.Lightning.GetInfo(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("unable to get local node info: %v", err)
-	}
-	nodeAddrs, err := parseNodeUris(info.Uris)
-	if err != nil {
-		return nil, fmt.Errorf("unable to parse node uris: %v", err)
+	params := &ServerOrderParams{}
+	bid, isBid := order.(*Bid)
+	isSidecar := isBid && bid.SidecarTicket != nil
+
+	// Using a sidecar ticket with a bid order means we won't be receiving
+	// the channel ourselves and are instead leasing a channel for another
+	// node. Therefore we need to add the other node's identity and multisig
+	// public key in the bid order we send to the auctioneer.
+	if isSidecar {
+		ticket := bid.SidecarTicket
+
+		// Make sure the sidecar ticket is in the correct state. It
+		// needs to have been registered with the recipient's node and
+		// that node's information must be present. If everything checks
+		// out, we add our signature over it since we are now sure that
+		// we have an order nonce set.
+		err := m.validateAndSignTicketForOrder(ctx, ticket, bid)
+		if err != nil {
+			return nil, fmt.Errorf("error validating sidecar "+
+				"ticket: %v", err)
+		}
+
+		copy(
+			params.NodePubkey[:],
+			ticket.Recipient.NodePubKey.SerializeCompressed(),
+		)
+		copy(
+			params.MultiSigKey[:],
+			ticket.Recipient.MultiSigPubKey.SerializeCompressed(),
+		)
+
+		// Most sidecar recipients won't be public nodes and therefore
+		// don't have connectable addresses.
+		//
+		// TODO(guggero): Add addresses to the Receiver part of the
```

Comment:
Yeah the only edge case here is if the maker on the other side is running _only_ over Tor. On the server-side if this becomes an issue, we can start to add information w.r.t preferred addresses for the acceptor. This way they can avoid nodes that are only Tor as that'll just result in failed execution.